nucleus/security/core/src/main/java/com/sun/enterprise/security/SecurityContext.java

```diff
@@ -72,11 +72,12 @@ public class SecurityContext extends AbstractSecurityContext {
 
     private static AuthPermission doAsPrivilegedPerm = new AuthPermission("doAsPrivileged");
 
+    // this is static because it's a thread local, which isn't serializable
+    private static ThreadLocal<Principal> sessionPrincipal = new ThreadLocal<>();
```

Comment:
Why did you remove `final`?

Reply:
You're right, @hs536 , thank you for spotting this.

I must have removed final when I was trying a different approach, but then forgot to put it back. I added it back now.